support/util/pausereconcile.go

```diff
@@ -16,38 +17,38 @@ const (
 )
 
 // IsReconciliationPaused checks the pauseUntil field to see if reconciliation on the resource should be
-// paused.
-func IsReconciliationPaused(logr logr.Logger, pausedUntilField *string) bool {
-	isReconciliationPaused, _ := processPausedUntilField(logr, pausedUntilField)
-	return isReconciliationPaused
+// paused and for how long.
+func IsReconciliationPaused(logr logr.Logger, pausedUntilField *string) (bool, time.Duration) {
+	isReconciliationPaused, _, pausedDuration := processPausedUntilField(logr, pausedUntilField)
+	return isReconciliationPaused, pausedDuration
 }
 
 // processPausedUntilField checks the pauseUntil field to see if reconciliation on the resource should be
-// paused. It can either be a date in RFC3339 format, or a boolean. If it is a date: reconciliation will be paused until
-// that date. If it is a true boolean value: it will be paused until the annotation is removed. If an improper value
+// paused. It can either be a date in RFC3339 format, or a Boolean. If it is a date: reconciliation will be paused until
+// that date. If it is a true boolean value: it will be paused until the value is removed. If an improper value
 // is specified: it will be treated as not paused. The second return boolean signifies if the user passed valid
-// input to the pausedUntil field.
-func processPausedUntilField(logr logr.Logger, pausedUntilField *string) (bool, bool) {
+// input to the pausedUntil field. The third return value is a duration of 0 or the difference between now and a date if it was given.
+func processPausedUntilField(logr logr.Logger, pausedUntilField *string) (bool, bool, time.Duration) {
```

Comment:
Can we name the return value? And using a `bool` to indicate validity seems weird to me
```suggestion
func processPausedUntilField(logr logr.Logger, pausedUntilField *string) (isPaused bool, validPauseConfig bool, pauseDuration time.Duration) {
```